controller/backup_volume_controller.go

```diff
@@ -0,0 +1,390 @@
+		log.Infof("Found %v backups in the backup target that do not exist in the backup target and need to be deleted", count)
+	}
+	for backupName := range backupsToDelete {
+		if err = bvc.ds.DeleteBackup(backupName); err != nil {
+			log.WithError(err).Errorf("Error deleting backup %s from cluster", backupName)
+		}
+	}
+
+	backupTarget, err := getDefaultBackupTarget(bvc.ds)
+	if err != nil {
+		log.WithError(err).Warnf("Backup target %s be deleted", defaultBackupTargetName)
+		return err
+	}
+
+	backupVolumeMetadataURL := backupstore.EncodeBackupURL("", backupVolumeName, backupTarget.Spec.BackupTargetURL)
+	configMetadata, err := backupTargetClient.GetConfigMetadata(backupVolumeMetadataURL)
+	if err != nil {
+		log.WithError(err).Error("Error getting backup volume config metadata from backup target")
+		// Ignore error to prevent enqueue
+		return nil
+	}
+	if configMetadata == nil {
+		return nil
+	}
+
+	updateLastSyncedAt := func() error {
+		backupVolume.Status.LastSyncedAt = &metav1.Time{Time: time.Now().UTC()}
+		if _, err = bvc.ds.UpdateBackupVolumeStatus(backupVolume); err != nil && !datastore.ErrorIsConflict(err) {
+			log.WithError(err).Error("Error updating backup volume status")
+			return err
+		}
+		return nil
+	}
+
+	// Check the backup volume config metadata got changed
+	if !backupVolume.Status.LastModificationTime.IsZero() &&
+		backupVolume.Status.LastModificationTime.Time.Equal(configMetadata.ModificationTime) {
+		return updateLastSyncedAt()
+	}
+
+	backupVolumeInfo, err := backupTargetClient.InspectBackupVolumeConfig(backupVolumeMetadataURL)
+	if err != nil {
+		log.WithError(err).Error("Error getting backup volume config from backup target")
+		// Ignore error to prevent enqueue
+		return nil
+	}
+	if backupVolumeInfo == nil {
+		return nil
+	}
+
+	// Updates BackupVolume CR status
+	backupVolume.Status.LastModificationTime = &metav1.Time{Time: configMetadata.ModificationTime}
+	backupVolume.Status.Size = backupVolumeInfo.Size
+	backupVolume.Status.Labels = backupVolumeInfo.Labels
+	backupVolume.Status.CreateAt = backupVolumeInfo.Created
+	backupVolume.Status.LastBackupName = backupVolumeInfo.LastBackupName
+	backupVolume.Status.LastBackupAt = backupVolumeInfo.LastBackupAt
+	backupVolume.Status.DataStored = backupVolumeInfo.DataStored
+	backupVolume.Status.Messages = backupVolumeInfo.Messages
+	backupVolume.Status.BackingImageName = backupVolumeInfo.BackingImageName
+	backupVolume.Status.BackingImageURL = backupVolumeInfo.BackingImageURL
+	return updateLastSyncedAt()
+}
+
+// cleanupBackups deletes related Backup CRs with the given backup volume name
+func (bvc *BackupVolumeController) cleanupBackups(backupVolumeName string, fileCleanupRequired bool) error {
+	backups, err := bvc.ds.ListBackupsWithBackupVolumeName(backupVolumeName)
+	if err != nil {
+		return err
+	}
+
+	var errs []string
+	if fileCleanupRequired {
+		for _, backup := range backups {
+			backup.Spec.FileCleanupRequired = true
```

Comment:
```
if !backup.Spec.FileCleanupRequired {
    // update
}
```

Reply:
If the backup volume `spec.FileCleanupRequired=true`, it means that all the corresponding backups need to set `spec.FileCleanupRequired=true`. This is for the case when the user clicks delete all backups of a backup volume.